AppCenter/AppCenter/Internals/DelegateForwarder/MSAppDelegateForwarder.m

```diff
@@ -0,0 +1,107 @@
+#import "MSAppDelegateForwarder.h"
+#import "MSCustomApplicationDelegate.h"
+#import "MSUtility+Application.h"
+
+static NSString *const kMSIsAppDelegateForwarderEnabledKey = @"AppCenterAppDelegateForwarderEnabled";
+
+// Original selectors with special handling.
+static NSString *const kMSOpenURLSourceApplicationAnnotation = @"application:openURL:sourceApplication:annotation:";
+static NSString *const kMSOpenURLOptions = @"application:openURL:options:";
+
+// Singleton instance.
+static MSAppDelegateForwarder *sharedInstance = nil;
+
+@implementation MSAppDelegateForwarder
+
+- (instancetype)init {
+  if ((self = [super init])) {
+#if !TARGET_OS_OSX
+    self.deprecatedSelectors = @{kMSOpenURLOptions : kMSOpenURLSourceApplicationAnnotation};
+#endif
+  }
+  return self;
+}
+
++ (instancetype)sharedInstance {
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [self new];
+  });
+  return sharedInstance;
+}
+
++ (void)resetSharedInstance {
+  sharedInstance = [self new];
+}
+
+- (Class)originalClassForSetDelegate {
+  return [MSApplication class];
+}
+
+#pragma mark - Custom Application
+
+- (void)custom_setDelegate:(id<MSApplicationDelegate>)delegate {
+
+  // Swizzle only once.
+  static dispatch_once_t delegateSwizzleOnceToken;
+  dispatch_once(&delegateSwizzleOnceToken, ^{
+    // Swizzle the delegate object before it's actually set.
```

Comment:
line break here

Reply:
clang format automatically removes the line break following the block syntax `^{` and there is no way to disable that behavior. 

I am fine to just conform with formatter here, but I am open to other suggestions.